utils/utils.go

```diff
@@ -44,6 +50,48 @@ func Min(a, b int) int {
 	return b
 }
 
+// DetermineNodename gets the node name, in order of priority:
+// 1. Hostname field in NetConf.
+// 2. Nodename field in NetConf (DEPRECATED).
+// 3. OS Hostname.
+func DetermineNodename(conf types.NetConf) string {
+	nodename, _ := os.Hostname()
+	if conf.Hostname != "" {
+		nodename = conf.Hostname
+		log.Warn("Configuration option 'hostname' is deprecated, use 'nodename' instead.")
+	}
+	if conf.Nodename != "" {
+		nodename = conf.Nodename
+	}
+	return nodename
+}
+
+// CreateOrUpdate tries to create the WEP first, if it already exists then it will try to Update it.
+// This is similar to what was previously called Apply operation.
+func CreateOrUpdate(ctx context.Context, client client.Interface, wep *api.WorkloadEndpoint) (*api.WorkloadEndpoint, error) {
```

Comment:
Shouldn't this logic be:

```
if wep.ResourceVersion:
  Update(wep)
else:
  Create(wep)
```

Reply:
The current logic ultimately does the same thing, so I think the logic is correct but your logic is a lot easier to read. I'll update it

Reply:
Well, the logic is actually different - the way I've suggested you don't need to perform the `Create` operation, whereas as it is now you always execute `Create` and only do an `Update` if `Create` fails.